src/main/java/com/teragrep/cfe_39/consumers/kafka/DatabaseOutput.java

```diff
@@ -0,0 +1,394 @@
+                            offset,                                     // 7 "offset", DataTypes.LongType
+                            UTF8String.fromBytes(origin).toString()     // 8 "origin", DataTypes.StringType
+                    );
+                    */
+
+                    // input
+                    final byte[] source = eventToSource();
+
+                    // origin
+                    final byte[] origin = eventToOrigin();
+
+                    // Format: Use AVRO format with syslog columns as indexed ones
+                    final long epochMicros = rfc3339ToEpoch(new RFC5424Timestamp(rfc5424Frame.timestamp).toZonedDateTime());
+                    SyslogRecord syslogRecord = SyslogRecord.newBuilder()
+                            .setTimestamp(epochMicros)
+                            .setMessage(rfc5424Frame.msg.toString())
+                            .setDirectory(rfc5424Frame.structuredData.getValue(teragrepDirectory).toString())
+                            .setStream(rfc5424Frame.structuredData.getValue(teragrepStreamName).toString()) // Or is sourcetype/stream supposed to be rfc5424Frame.appName.toString() instead?
+                            .setHost(rfc5424Frame.hostname.toString())
+                            .setInput(new String(source, StandardCharsets.UTF_8))
+                            .setPartition(recordOffsetObject.getPartition().toString())
+                            .setOffset(recordOffsetObject.getOffset())
+                            .setOrigin(new String(origin, StandardCharsets.UTF_8))
+                            .build();
+
+
+                    // Calculate the size of syslogRecord that is going to be written to syslogAvroWriter-file.
+                    long capacity = syslogRecord.toByteBuffer().capacity();
+                    // handle initial timestamp
+                    if (epochMicros_last == 0L) {
+                        epochMicros_last = epochMicros;
+                    }
+                    // Check if there is still room in syslogAvroWriter for another syslogRecord. Commit syslogAvroWriter to HDFS if no room left, emptying it out in the process.
+                    // boolean fileCommitted = committedToHdfs(approximatedSize + capacity, lastObject); // FIXME: approximatedSize is not working properly without the use of flush() after append. File sizes are all over the place.
+                    boolean fileCommitted = committedToHdfs(syslogAvroWriter.getFileSize() + capacity, (RecordOffset) lastObject);
+                    if (fileCommitted) {
+                        LOGGER.debug("Target file size reached, file <{}> stored to <{}> in HDFS", syslogFile.getName(), lastObject.getTopic()+"/"+lastObject.getPartition()+"."+lastObject.getOffset());
+                    }else {
+                        LOGGER.debug("Target file size not yet reached, continuing writing records to <{}>.", syslogFile.getName());
+                    }
```

Comment:
Function calls in logging statements should be guarded with `if(LOGGER.isDebugEnabled())` and such

Reply:
Solved in commit b12bb1423c085c18f5152a9459b0caec25db8457